ipyleaflet/leaflet.py

```diff
@@ -1932,10 +1932,13 @@ class LegendControl(Control):
         "value 2": "#55A",
         "value 3": "#005"}).tag(sync=True)
 
-    def __init__(self, legend, *args, name="Legend", **kwargs):
+    def __init__(self, legend, *args, **kwargs):
+        kwargs["legend"] = legend
+        # For backwards compatibility with ipyleaflet<=0.16.0
+        if 'name' in kwargs:
+            kwargs.setdefault('title', kwargs['name'])
```

Comment:
Should we show a warning in this case?

Reply:
We could, but I'm also happy to support it as an undocumented option for now for backwards compatibility. We might ask ourselves why we don't make the other changes just deprecations at this point too, and remove them at some later point, to ease backwards compatibility.

Do you want it deprecated?

Reply:
I added a bunch of deprecation warnings, and added back the removed apis to be deprecated. I also changed the pendingdeprecationwarnings to just deprecation warnings.